docs/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to.md

```diff
@@ -0,0 +1,518 @@
+
+<!-- For information about which features might be added in future releases, see the [Roadmap](https://github.com/dotnet/runtime/tree/master/src/libraries/System.Text.Json/roadmap/README.md). [Restore this when the roadmap is updated.]-->
+
+Most of this article is about how to use the <xref:System.Text.Json.JsonSerializer> API, but it also includes guidance on how to use the <xref:System.Text.Json.JsonDocument> Document Object Model (DOM) API and the <xref:System.Text.Json.Utf8JsonReader> and <xref:System.Text.Json.Utf8JsonWriter> API. The article is organized into sections in the following order:
+
+* Differences in default `System.Text.Json.JsonSerializer` behavior compared to Newtonsoft.
+* Scenarios that `JsonSerializer` doesn't support, but workarounds might be feasible. [Go to the first of these sections.](#deserialize-inferred-types-to-object-properties)
+* Scenarios that `JsonSerializer` doesn't support, and workarounds are relatively difficult or impractical. [Go to the first of these sections.](#types-without-built-in-support).
+* [Utf8JsonReader and Utf8JsonWriter](#utf8jsonreader-and-utf8jsonwriter).
+* [JsonDocument](#jsondocument).
+
+## Case-insensitive deserialization 
+
+During deserialization, Newtonsoft does case-insensitive property name matching by default. The `System.Text.Json` default is case-sensitive, which gives better performance. For information about how to do case-insensitive matching, see [Case-insensitive property matching](system-text-json-how-to.md#case-insensitive-property-matching).
+
+If you're using `System.Text.Json` indirectly by using ASP.NET Core, you don't need to do anything to get behavior like Newtonsoft. ASP.NET Core specifies the case-insensitive setting when it uses `System.Text.Json`.
+
+## Comments
+
+During deserialization, Newtonsoft ignores comments in the JSON by default. The `System.Text.Json` default is to throw exceptions for comments because the [RFC 8259](https://tools.ietf.org/html/rfc8259) specification doesn't include them. For information about how to allow comments, see [Allow comments and trailing commas](system-text-json-how-to.md#allow-comments-and-trailing-commas).
+
+## Trailing commas
+
+During deserialization, Newtonsoft ignores trailing commas by default. In some cases, it ignores multiple trailing commas (for example, `[{"Color":"Red"},{"Color":"Green"},,]`). The `System.Text.Json` default is to throw exceptions for trailing commas because the [RFC 8259](https://tools.ietf.org/html/rfc8259) specification doesn't include them. For information about how to allow trailing commas, see [Allow comments and trailing commas](system-text-json-how-to.md#allow-comments-and-trailing-commas). There's no way to allow multiple trailing commas.
+
+## Character escaping
```

Comment:
Should this be moved further down the list? ASP.NET Core configures this to use the less aggressive encoding behavior, so we've seen no issue reports for this. On the other hand, we've had lots of users confused by the strict date time parsing and "Quoted numbers" changes. Might be useful to bump those further up?

Reply:
At present the sections are in groups: no workaround needed, workaround sample code provided, and no workaround sample code provided. If I keep that organization I can move encoding  to the end of the first group and move quoted numbers and dates to the start of the second group. If you think that still leaves encoding too prominent or the others too far down, I'm open to suggestions for changing the organization.

Reply:
Let's start with what it's already here, maybe we can tweak it based on if users have difficulty discovering things.